pandas/tests/series/methods/test_convert_dtypes.py

```diff
@@ -144,11 +156,23 @@ class TestSeriesConvertDtypes:
                 [1, 2.0],
```

Comment:
It's a bit hard to interpret below tests / diff, but does pd.Series([1, 2.0], dtype=object).convert_dtypes() still give Int64?

Reply:
Yes.  Here's the interpretation of the tests for that case:
The code reads:
```python
            (
                [1, 2.0],
                object,
                {
                    ((True,), (True, False), (True,), (True, False)): "Int64",
                    ((True,), (True, False), (False,), (True, False)): np.dtype(
                        "float"
                    ),
                    ((False,), (True, False), (True, False), (True, False)): np.dtype(
                        "object"
                    ),
                },
            ),
```
This means the following:
1) Create a `Series` with `[1, 2.0]` as the entries, with dtype `object`
2) Consider the 16 possible combinations of the 4 arguments `infer_objects`, `convert_string`, `convert_integer` and `convert_boolean`
3) If `infer_objects==True` and `convert_integer==True`, result should be `Int64`
4) If `infer_objects==True` and `convert_integer==False`, result should be `float`
5) If `infer_objects==False`, result is always `object`

Prior to this PR, the tests were as follows:
p3) If `convert_integer==True`, result should be `Int64` independent of value of `infer_objects`
p4) If `infer_objects==True` and `convert_integer==False`, result should be `float` (same)
p5) If `infer_objects==False` and `convert_integer==False`, result is `object`

I think the new version is what we want the behavior to be, i.e., if you start with `object` and  you don't do the infer-objects step, it remains an object.

Reply:
OK, thanks